agent/app/agent_compatibility_linux.go

```diff
@@ -0,0 +1,48 @@
+// +build linux
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package app
+
+import (
+	"github.com/aws/amazon-ecs-agent/agent/config"
+	"github.com/aws/amazon-ecs-agent/agent/engine"
+	"github.com/aws/amazon-ecs-agent/agent/utils"
+	"github.com/cihub/seelog"
+	"github.com/pkg/errors"
+)
+
+func (agent *ecsAgent) checkCompatibility(engine engine.TaskEngine) error {
+	tasks, err := engine.ListTasks()
+	if err != nil {
+		return err
+	}
+
+	var incompatible bool
+	for _, task := range tasks {
+		if utils.ZeroOrNil(task.MemoryCPULimitsEnabled) {
```

Comment:
Why are we checking for `task.MemoryCPULimitsEnabled` here? Shouldn't the mere fact that there tasks that are already running on the Agent prevent us from enabling this feature? As otherwise, there'd be a skewed `cgroup` hierarchy?

Reply:
If we are restoring from a snapshot of the current or later agent version, with limits enabled, its completely reasonable to proceed.